pulp_rpm/app/tasks/publishing.py

```diff
@@ -262,40 +260,33 @@ def create_repomd_xml(packages, publication, extra_repomdrecords, sub_folder=Non
         oth_db.add_pkg(pkg)
 
     # Process update records
-    for update_record in UpdateRecord.objects.filter(
-            pk__in=publication.repository_version.content):
+    for update_record in UpdateRecord.objects.filter(pk__in=contents).iterator():
```

Comment:
that's the reason why I exposed `contents` instead of `packages`. Before, this and the other queries below were only pointing to main repo contents. It was a bug for kickstarts, because sub repo contents were not exposed, so this query would never find the desired sub repo content